src/algebra/quaternion.lean

```diff
@@ -423,42 +392,33 @@ section char_zero
 variables [no_zero_divisors R] [char_zero R]
 
 @[simp]
-lemma conj_eq_self {c₁ c₂ : R} {a : ℍ[R, c₁, c₂]} :
-  conj a = a ↔ a = a.re :=
+lemma star_eq_self {c₁ c₂ : R} {a : ℍ[R, c₁, c₂]} :
+  star a = a ↔ a = a.re :=
 by simp [ext_iff, neg_eq_iff_add_eq_zero, add_self_eq_zero]
 
-lemma conj_eq_neg {c₁ c₂ : R} {a : ℍ[R, c₁, c₂]} :
-  conj a = -a ↔ a.re = 0 :=
+lemma star_eq_neg {c₁ c₂ : R} {a : ℍ[R, c₁, c₂]} :
+  star a = -a ↔ a.re = 0 :=
 by simp [ext_iff, eq_neg_iff_add_eq_zero]
 
 end char_zero
--- Can't use `rw ← conj_eq_self` in the proof without additional assumptions
-
-lemma conj_mul_eq_coe : conj a * a = (conj a * a).re := by ext; simp; ring_exp
-
-lemma mul_conj_eq_coe : a * conj a = (a * conj a).re :=
-by { rw a.commute_self_conj.eq, exact a.conj_mul_eq_coe }
+-- Can't use `rw ← star_eq_self` in the proof without additional assumptions
 
-lemma conj_zero : conj (0 : ℍ[R, c₁, c₂]) = 0 := conj.map_zero
+lemma star_mul_eq_coe : star a * a = (star a * a).re := by ext; simp; ring_exp
 
-lemma conj_neg : (-a).conj = -a.conj := (conj : ℍ[R, c₁, c₂] ≃ₗ[R] _).map_neg a
-
-lemma conj_sub : (a - b).conj = a.conj - b.conj := (conj : ℍ[R, c₁, c₂] ≃ₗ[R] _).map_sub a b
-
-
-@[simp] lemma conj_pow (n : ℕ) : (a ^ n).conj = a.conj ^ n := star_pow _ _
+lemma mul_star_eq_coe : a * star a = (a * star a).re :=
+by { rw ←star_comm_self', exact a.star_mul_eq_coe }
 
 open mul_opposite
 
 /-- Quaternion conjugate as an `alg_equiv` to the opposite ring. -/
-def conj_ae : ℍ[R, c₁, c₂] ≃ₐ[R] (ℍ[R, c₁, c₂]ᵐᵒᵖ) :=
-{ to_fun := op ∘ conj,
-  inv_fun := conj ∘ unop,
+def star_ae : ℍ[R, c₁, c₂] ≃ₐ[R] (ℍ[R, c₁, c₂]ᵐᵒᵖ) :=
```

Comment:
Do we have a generic version of this definition? If not, should we?

Reply:
Otherwise LGTM (and we can leave this generalization till after the port).

Reply:
bors d+

Reply:
I don't think we do; we have a typeclass that says `star (c • x) = star c • star x` but not one for `star (c • x) = c • star x`